### ISSUE TITLE
Add adaptive learning rate for gradient descent

### ISSUE THREAD

**Issue by mikeheddes** [mikeheddes/dcgp.js#29]
Use a form of line search to make the learning rate adaptive.

**Comment by mikeheddes** [comment]
Closed with #32
